gdmix-workflow/src/gdmixworkflow/distributed_workflow.py

```diff
@@ -18,26 +18,23 @@ def gdmix_distributed_workflow(gdmix_config_file, namespace, secret_name, image,
     current_op = no_op("GDMix-training-start")
     suffix = gen_random_string()
 
-    if hasattr(gdmix_config_obj, FIXED_EFFECT_CONFIG):
-        fe_tip_op = no_op("fixed-effect-training-start")
-        fe_tip_op.after(current_op)
-        fe_workflow = FixedEffectWorkflowGenerator(gdmix_config_obj,
-                                                   namespace=namespace,
-                                                   secret_name=secret_name,
-                                                   image=image,
-                                                   service_account=service_account,
-                                                   job_suffix=suffix)
-        fe_start_op, current_op = fe_workflow.gen_workflow()
-        fe_start_op.after(fe_tip_op)
+    if not hasattr(gdmix_config_obj, FIXED_EFFECT_CONFIG):
+        raise ValueError(f"Need to define {FIXED_EFFECT_CONFIG}")
```

Comment:
Fixed effect is optional, user can train random effect only.

Reply:
In your code, you had this hidden dependency. I just made it explicit. (The old code reference the fixed effect config from the random effect to access the name).
I prefer no to change the logic behind the science here and maybe you can fix it later.

Reply:
okay.